### ISSUE TITLE
Proposal: Make short option is optional

### ISSUE THREAD

**Issue by KrauseFx** [fastlane/fastlane#4136]
```ruby
      if type.to_s.length > 0 and short_option.to_s.length == 0
        UI.user_error!("Type '#{type}' for key '#{key}' requires a short option")
      end
```

```
/Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/fastlane_core-0.41.2/lib/fastlane_core/ui/interface.rb:139:in `user_error!': Type 'String' for key 'api_token' requires a short option (FastlaneCore::Interface::FastlaneError)
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/fastlane_core-0.41.2/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	from /Users/fkrause/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/fastlane_core-0.41.2/lib/fastlane_core/configuration/config_item.rb:32:in `initialize'
	from /Users/fkrause/Developer/fastlane/fastlane/lib/fastlane/actions/appetize.rb:90:in `new'
	from /Users/fkrause/Developer/fastlane/fastlane/lib/fastlane/actions/appetize.rb:90:in `available_options'
	from /Users/fkrause/Developer/fastlane/fastlane/spec/unused_options_spec.rb:7:in `block (4 levels) in <top (r
```

As @lacostej pointed out, this is not right. In particular this is also required when building a fastlane action (not a CLI) and it would require a short option. 

I'd like to remove this `user_error!` call, I just wanted to be sure I didn't forget anything.

cc @mfurtak @lacostej

**Comment by lacostej** [comment]
@KrauseFx I am not sure why we require `short_options`. I don't see the need to enforce it. Especially as it isn't enforced for Booleans today.

Note that removing this code would fix the rubocop warnings on my PR #4142 :)

**Comment by mfurtak** [comment]
I don't know the origin of requiring short options, but I remember it being described to me as working around a problem with Commander.

However, since then, I believe I fixed the bit that was causing problems if short options were not provided. See this commit:

https://github.com/fastlane/fastlane/commit/7f4187d628d175375a5d460d98d7b1b2bd36fa1d

Given that, I think it's OK to make them optional. 👍

**Comment by TKBurner** [comment]
@lacostej Let us know if you have any thoughts before we consider this as a team 👍

**Comment by lacostej** [comment]
@TKBurner I already answered in https://github.com/fastlane/fastlane/issues/4136#issuecomment-207759078 

I don't see the need to enforce it. Remove those 3 lines! :+1:

**Comment by lacostej** [comment]
@TKBurner @mfurtak @KrauseFx  will this be merged ?

**Comment by TKBurner** [comment]
Thanks @lacostej as always really really appreciate what you are doing by working with us 👍

**Comment by mfurtak** [comment]
Addressed in #4292
